Ship_Game/EmpireResearch.cs

```diff
@@ -182,5 +181,158 @@ public void ReorderTech(int oldIndex, int newIndex)
         {
             Queue.Reorder(oldIndex, newIndex);
         }
+        
+        
+        /// <summary>
+        /// Removes the given tech from the queue, and all techs that depend on it.
+        /// </summary>
+        public void RemoveTechFromQueue(string techUid)
+        {
+            void RemoveLeadsToRecursive(string tech)
+            {
+                Queue.Remove(tech);
+                foreach (Technology.LeadsToTech dependent in ResourceManager.Tech(tech).LeadsTo)
+                    RemoveLeadsToRecursive(dependent.UID);
+            }
+
+            RemoveLeadsToRecursive(techUid);
+        }
+        
+        
+        /// <summary>
+        /// Adds the given tech to the queue, and all of its PreReqs.
+        /// It checks if the tech (and PreReqs) have been discovered and not finished yet and not already enqueued.
+        /// </summary>
+        public void AddTechToQueue(string techUid)
+        {
+            var technology = ResourceManager.Tech(techUid);
+            var predecessorTechs = technology.PredecessorTechs();
+            foreach (Technology predecessor in predecessorTechs)
+            {
+                var techEntry = Empire.GetTechEntry(predecessor.UID);
+                if (techEntry.Discovered && !techEntry.Unlocked && !IsQueued(predecessor.UID))
+                {
+                    AddToQueue(predecessor.UID);
+                }
+            }
+            AddToQueue(techUid);
+        }
+        
+        
+        string ResearchUidAt(int index) => Queue[index];
+        Technology ResearchTechnologyAt(int index) => ResourceManager.Tech(ResearchUidAt(index));
+        public bool AboveIsPreReq(int currentIndex)
+        {
+            var currentTech = ResearchTechnologyAt(currentIndex);
+            foreach (Technology.LeadsToTech dependent in ResourceManager.Tech(ResearchUidAt(currentIndex - 1)).LeadsTo)
+                if (dependent.UID == currentTech.UID)
+                    return true;
+            return false;
+        }
+        
+        public bool IsPreReqOfBellow(int currentIndex)
+        {
+            Technology current = ResearchTechnologyAt(currentIndex);
+            string next = ResearchUidAt(currentIndex + 1);
+            foreach (Technology.LeadsToTech dependent in current.LeadsTo)
+                if (dependent.UID == next)
+                    return true;
+            return false;
+        }
+        
+        private void SwapQueueItems(int first, int second)
+        {
+            (Queue[first], Queue[second]) = (Queue[second], Queue[first]);
+        }
+        
+        public void MoveUp(int index)
+        {
+            if (ResearchCanMoveUp(index))
+            {
+                SwapQueueItems(index - 1, index);
+            }
```

Comment:
If you have an If-then block with just one line, you can do the following:
Consider doing it for the rest of your changes
```suggestion
            if (ResearchCanMoveUp(index))
                SwapQueueItems(index - 1, index);
```